### PR TITLE
Modern Packaging

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Changes

- Added some basic CI workflows
- Replaced `setuptools` with `flit` and `pyproject.toml`
- Added an `AUTHORS.rst` and a `bump2version` configuration (under `setup,cfg`)
- Linting with `ruff`
- Made testing safer with non-POSIX environments
- Martin, Carsten, and I are now listed as authors/maintainers

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+[build-system]
+build-backend = "flit_core.buildapi"
+requires = ["flit_core >=3.8,<4"]
+
+[project]
+name = "roocs_grids"
+keywords = ["roocs_grids", "regrid", "grid", "interpolate", "cmip"]
+authors = [
+  {name = "Ag Stephens", email = "nielszeilemaker@godatadriven.com"},
```
Comment (sol1105) [inline pyproject.toml]: This mail address is a remnant from the template I guess

**Comment by sol1105** [comment]
@Zeitsperre Thanks a lot, that's very helpful :-)

File: pyproject.toml (hunk 1 above)
Comment (Zeitsperre) [reply pyproject.toml]: Woops, yes, haha. Will fix.

**Comment by sol1105** [comment]
@Zeitsperre @cehbrecht 
I renamed the `aux` directory to `scripts` and moved it back into `roocs_grids`. Do you think this is reintroducing the `clisops` circular dependency issue?
Alternatively I could think of creating a `docs` folder with proper documentation and somehow integrating the `scripts/*` in there. But maybe for now this will do?

I updated the `HISTORY.rst` but I am not sure if this changes can still be counted as bug fixes, so one could also release it as `0.2.0` instead of `0.1.2`?

**Comment by cehbrecht** [comment]
> I renamed the `aux` directory to `scripts` and moved it back into `roocs_grids`. Do you think this is reintroducing the `clisops` circular dependency issue?

@sol1105 I would move the `scripts` folder to the top-level folder ... not being part of the roocs_grids source. We could also exclude it from packaging in the `MANIFEST.in`.

**Comment by sol1105** [comment]
@cehbrecht @Zeitsperre 

> > I renamed the `aux` directory to `scripts` and moved it back into `roocs_grids`. Do you think this is reintroducing the `clisops` circular dependency issue?
> 
> @sol1105 I would move the `scripts` folder to the top-level folder ... not being part of the roocs_grids source. We could also exclude it from packaging in the `MANIFEST.in`.

I will then move it to the top-level folder. It then should be excluded from packaging automatically (edit: it wasn't). In future releases I can create a docs section that includes at least the python script and explains how to use `roocs_grids` in combination with `clisops` / `xarray`. Please let me know, if you prefer some other approach.

**Comment by cehbrecht** [comment]
> I will then move it to the top-level folder. It then should be excluded from packaging automatically (edit: it wasn't). In future releases I can create a docs section that includes at least the python script and explains how to use `roocs_grids` in combination with `clisops` / `xarray`. Please let me know, if you prefer some other approach.

fine for me 👍

**Comment by Zeitsperre** [comment]
This looks good from my end as well!

**Pull request merged**